validator/js/engine/parse-srcset.js

```diff
@@ -69,22 +69,37 @@ const parseSrcset = function(srcset) {
   // more spaces + a non empty string containing no space or commas).
   // Doesn't capture the initial space.
   //
-  // \s*                       Match, but don't capture leading spaces
+  // \s*                       Match, but don't capture leading spaces.
   // (?:,\s*)?                 Optionally match comma and trailing space,
   //                           but don't capture comma.
   // ([^,\s]\S*[^,\s])         Match something like "google.com/favicon.ico"
-  //                           but not ",google.com/favicon.ico,"
+  //                           but not ",google.com/favicon.ico,".
   // \s*                       Match, but dont capture spaces.
-  // ([\d]+.?[\d]*[w|x])?      e.g. "5w" or "5x" or "10.2x"
-  // \s*                       Match, but don't capture space
+  // (                         Match the width or density descriptor...
+  //   [1-9]\d*[wx]            which is a non-zero integer followed by a w
+  //                           or an x ...
+  //   |                       or ...
+  //   [1-9]\d*\.\d+x          a float with its integer greater than zero and
+  //                           followed by an x ...
+  //   |                       or ...
+  //   0.\d*[1-9]\d*x          a float with its fractional part greater than
+  //                           zero and followed by an x ...
+  // )?                        and make it optional.
+  // \s*                       Match, but don't capture space.
```

Comment:
The comment describing the regex was difficult to fit into the line width when taking the new dimension match as a whole, so I split it up into its three variations, but I was unsure how best to format this now.